docs/examples/fd_fitting_intermediate/index.rst

```diff
@@ -0,0 +1,374 @@
+model for a particular model parameter. Let's have a look at the parameters it needs::
+
+    >>> help(pylake.parameter_trace)
+
+    Help on function parameter_trace in module lumicks.pylake.fitting.detail.parameter_trace:
+
+    parameter_trace(model, parameters, inverted_parameter, independent, dependent, **kwargs)
+        Invert a model with respect to one parameter. This function fits a unique parameter for every data point in
+        this data-set while keeping all other parameters fixed. This can be used to for example invert the model with
+        respect to the contour length or some other parameter.
+
+        Parameters
+        ----------
+        model : Model
+            Fitting model.
+        parameters : Parameters
+            Model parameters.
+        inverted_parameter : str
+            Parameter to invert.
+        independent : array_like
+            vector of values for the independent variable
+        dependent: array_like
+            vector of values for the dependent variable
+        **kwargs:
+            forwarded to scipy.optimize.least_squares
+
+        Examples
+        --------
+        ::
+            # Define the model to be fitted
+            model = pylake.inverted_odijk("model") + pylake.force_offset("model")
+
+            # Fit the overall model first
+            data_handle = model.add_data("dataset1", f=force_data, d=distance_data)
+            current_fit = pylake.Fit(model)
+            current_fit.fit()
+
+            # Calculate a per data point contour length
+            lcs = parameter_trace(model, current_fit[data_handle], "model/Lc", distance, force)
+
+Let's see if we have all these pieces of information. Our model was called `m_dna`. We can extract
+the parameters for the RecA condition using the name we provided to the dataset before (i.e. `fit["RecA"]`).
+The parameter we wish to invert for is `DNA/Lc` and for the independent and dependent variables we simply
+pass the dataset::
+
+    Lcs = pylake.parameter_trace(m_dna, fit["RecA"], 'DNA/Lc', distance_dynamic, force_dynamic)
+
+Let's plot it::
+
+    plt.plot(Lcs)
+    plt.ylabel('Contour lengths')
+    plt.xlabel('Time [s]')
+
+.. image:: output_10_7.png
+
+Looks like some of the estimates are way off early in the curve. Doing this inversion at very low
+distances is quite error prone, likely due to the non-linearity of the model. In addition, the Odijk
+model is known to not be reliable at low forces, so we would like to exclude this data anyway. Let's
+only look at the points where the distance is higher than 2.25::
+
+    distance_mask = distance_dynamic > 2.2
+
+    plt.plot(distance_dynamic[distance_mask], Lcs[distance_mask])
+    plt.ylabel('Contour length [micron]')
+    plt.xlabel('Distance [micron]')
+
+.. image:: output_10_8.png
+
+Here we can see the different contour length transitions quite clearly. There seems to be one region
+of contour lengths around 3.2 before finally lengthening to 3.4 micrometers.
```

Comment:
Comment for the overall page: I'm curious about what BD will say, but to me this looks like a very nice example. Well written and it nicely walks the user through the fitting process and various data-specific challenges.

Reply:
I have just added the suggestions from BD to the documentation. Tomorrow I will try to clean up the history a bit and undo the FD => Fd bit for `FDCurve` and `DownsampledFD`.

I'd prefer keeping the `import lumicks.pylake as lk` for the examples, since they share the same header. How do you feel about this?

So how strongly do you think this commit history should be squashed?

On March 24/25 I reconstructed the history up to that point and made the history fairly clean (even though it is a big chunk of commits). Shall I clean up the history post 24th grouping the changes?